pyquil/operator_estimation.py

```diff
@@ -799,17 +850,169 @@ def measure_observables(qc: QuantumComputer, tomo_experiment: TomographyExperime
                 )
                 continue
 
-            # 3.4 Pick columns corresponding to qubits with a non-identity out_operation and stack
-            #     into an array of shape (n_shots, n_measure_qubits)
-            my_obs_strings = np.vstack(obs_strings[q] for q, op_str in setting.out_operator).T
-
-            # 3.6 Multiply row-wise to get operator values. Do statistics. Yield result.
-            obs_vals = coeff * np.prod(my_obs_strings, axis=1)
-            obs_mean = np.mean(obs_vals)
-            obs_var = np.var(obs_vals) / n_shots
-            yield ExperimentResult(
-                setting=setting,
-                expectation=obs_mean.item(),
-                stddev=np.sqrt(obs_var).item(),
-                total_counts=n_shots,
-            )
+            # 3.3 Obtain statistics from result of experiment
+            obs_mean, obs_var = _stats_from_measurements(bitstrings, setting, n_shots, coeff)
+
+            if calibrate_readout:
```

Comment:
this *whole* block is duplicated code and should be factored out

Reply:
This is not really duplicated code, at least not one that can be easily factored out without also giving up significantly on readability. Compare this code block to e.g. the one in steps 1.1 - 1.3, or those in 1.4 - 2.1. There are enough tiny but significant differences that those deserve to live in separate code blocks, and not just callable through some common function.